src/libAtomVM/term.h

```diff
@@ -879,6 +881,29 @@ static inline int term_binary_data_size_in_terms(uint32_t size)
     }
 }
 
+#if TERM_BYTES == 4
+#define TERM_BINARY_DATA_SIZE_IN_TERMS(size) \
+  TERM_BINARY_SIZE_IS_HEAP_SIZE(size) ? (((size) + 4 - 1) >> 2) + 1 : TERM_BOXED_REFC_BINARY_SIZE
+#elif TERM_BYTES == 8
+#define TERM_BINARY_DATA_SIZE_IN_TERMS(size) \
+  TERM_BINARY_SIZE_IS_HEAP_SIZE(size) ? (((size) + 8 - 1) >> 3) + 1 : TERM_BOXED_REFC_BINARY_SIZE
+#endif
+
+/**
+ * @brief The total mount of heap space used by a binary (including headers)
+ *
+ * @details Returns the heap size of a binary.
+ * @param size the size in bytes
+ * @return the heap size in terms
+ */
+static inline int term_binary_heap_size(uint32_t size)
```

Comment:
Is this function still used?

Macros were created to please compiler about avoiding dynamic size array in C-stack-based heap allocation, but do we want to keep the functions?

Reply:
Good question.  I added this as a gesture to remove the `+ BINARY_HEADER_SIZE` pattern that litters the codebase, but maybe that should be a macro, as well.

Reply:
Removed function and updated the macros